src/lib/snark_work_lib/selector.ml

```diff
@@ -0,0 +1,82 @@
+open Core_kernel
+
+module Single = struct
+  module Spec = struct
+    [%%versioned
+    module Stable = struct
+      [@@@no_toplevel_latest_type]
+
+      module V1 = struct
+        type t =
+          ( Transaction_witness.Stable.V2.t
+          , Ledger_proof.Stable.V2.t )
+          Work.Single.Spec.Stable.V2.t
+        [@@deriving sexp, yojson]
+
+        let to_latest = Fn.id
+      end
+    end]
+
+    type t = (Transaction_witness.t, Ledger_proof.Cached.t) Work.Single.Spec.t
+
+    let materialize : t -> Stable.Latest.t =
```

Comment:
Let's stick to `read_all_proofs_from_disk`, this is the convention we use across the codebase.

Reply:
I will do. But this name is too long and I called this all the time, so I'm a bit unsatisfied with the naming.